GeoInterfaceMakie/src/GeoInterfaceMakie.jl

```diff
@@ -23,12 +23,30 @@ function _convert_arguments(t, geom)::Tuple
     geob = GI.convert(GB, geom)
     MC.convert_arguments(t, geob)
 end
+
 function _convert_array_arguments(t, geoms::AbstractArray{T})::Tuple where T
     if Missing <: T
         geob = map(geom -> GI.convert(GB, geom), skipmissing(geoms))
     else
         geob = map(geom -> GI.convert(GB, geom), geoms)
     end
+    if !(eltype(geob) <: GB.AbstractGeometry) || eltype(geob) isa Union # Unions are bad
+        if isempty(geob)
+            geob = geob
+        end
+        first_trait = GI.geomtrait(first(geob))
+        different_trait_idx = findfirst(x -> GI.geomtrait(x) != first_trait, geob)
+        if isnothing(different_trait_idx)
+            # All traits are the same, and all things are GB, so how did this happen?
+            # It's possible that the point lengths are different, we still need to handle that.
+            return MC.convert_arguments(t, geob)
+        end
+        if all(x -> GI.geomtrait(x) isa Union{GI.MultiPolygonTrait, GI.PolygonTrait, GI.GeometryCollectionTrait}, geob) # an array of polygon like structs
+            geob = to_multipoly(geob)
```

Comment:
Might generate better code if you return here

Reply:
```suggestion
            return to_multipoly(geob)